internal/server/config.go

```diff
@@ -734,8 +752,7 @@ func (s Server) loadIdentities(db *gorm.DB, identities []Identity) error {
 		keep = append(keep, identity.ID)
 	}
 
-	// remove any provider previously defined by config
-	if err := data.DeleteIdentities(db, data.NotIDs(keep), data.CreatedBy(models.CreatedBySystem)); err != nil {
+	if err := data.DeleteIdentities(db, data.NotIDs(keep)); err != nil {
```

Comment:
Is this missing a filter for "from infra provider" ? Otherwise we lose all identities from other providers. That would be a good test case, if we don't have it already.

Reply:
In this case we'll still want to delete all – since using this config implies Infra will converge to these users existing in the system (vs leaving users lying around).